### PR TITLE
Add L1 tests for signing

### PR DESCRIPTION
**Pull request by stephenmichaelf**
This change adds L1 tests for task signature verification.

There is also some refactoring to the L1 test setup that allows us to modify mocked values before running tests.

### CODE REVIEW

File: src/Test/L1/Worker/L1TestBase.cs [hunk 1]
```diff
@@ -93,8 +93,7 @@ protected static TaskStep CreateScriptTask(string script)
             return step;
         }
 
-        protected async Task<TestResults> RunWorker(Pipelines.AgentJobRequestMessage message,
-            [CallerMemberName] string testName = "")
+        public void SetupL1([CallerMemberName] string testName = "")
```
Comment (stephenmichaelf) [inline src/Test/L1/Worker/L1TestBase.cs]: Added a separate `SetupL1` method, similar to what we have in the L0 tests. This allows us to set things up, adjust mocked values, then execute the test.

File: src/Test/L1/Worker/L1TestBase.cs [hunk 2]
```diff
@@ -108,21 +107,29 @@ protected static TaskStep CreateScriptTask(string script)
             var stringFile = Path.Combine(assemblyLocation, "en-US", "strings.json");
             StringUtil.LoadExternalLocalization(stringFile);
 
-            using (L1HostContext context = new L1HostContext("Agent", GetLogFile(this, testName)))
```
Comment (alex-peck) [inline src/Test/L1/Worker/L1TestBase.cs]: We probably should still dispose the `L1HostContext` somewhere.

Comment (stephenmichaelf) [reply src/Test/L1/Worker/L1TestBase.cs]: Add disposal after tests.

File: src/Test/L1/Worker/L1TestBase.cs [hunk 3]
```diff
@@ -108,21 +107,29 @@ protected static TaskStep CreateScriptTask(string script)
...
-                // Use different working directories for each test
-                var config = GetMockedService<FakeConfigurationStore>();
-                config.WorkingDirectoryName = testName;
+            // Use different working directories for each test
+            var config = GetMockedService<FakeConfigurationStore>(); // TODO: Need to update this. can hack it for now.
+            config.WorkingDirectoryName = testName;
+        }
 
-                await SetupMessage(context, message);
+        protected L1HostContext _l1hc;
```
Comment (jpricket) [inline src/Test/L1/Worker/L1TestBase.cs]: nit: not a fan of abbreviated variable names. hostContext or l1hostcontext might be more readable.

File: src/Test/L1/Worker/WorkerL1Tests.cs [hunk 4]
```diff
@@ -34,6 +36,7 @@ public async Task Test_Base()
             {
                 Assert.Equal(expectedSteps[idx], steps[idx].Name);
             }
+            TearDown();
```
Comment (jpricket) [inline src/Test/L1/Worker/WorkerL1Tests.cs]: If disposing the l1hostcontext is really important, TearDown should be in a finally. If the assert fails, teardown won't happen. This is true for all the test methods. You may want to create a RunTest method takes the "action" to test and does the setup and tear down.

Comment (stephenmichaelf) [reply src/Test/L1/Worker/WorkerL1Tests.cs]: Added try/finally to stay consistent with other parts of the code.

File: src/Test/L1/Worker/L1TestBase.cs (hunk 3 above)
Comment (stephenmichaelf) [reply src/Test/L1/Worker/L1TestBase.cs]: Renamed to `_l1HostContext`.

**Pull request merged**